manifests/forwarder.pp

```diff
@@ -51,43 +55,50 @@
 # Requires: nothing
 #
 class splunk::forwarder (
-  $server                 = $splunk::params::server,
-  $package_source         = $splunk::params::forwarder_pkg_src,
-  $package_name           = $splunk::params::forwarder_pkg_name,
-  $package_ensure         = $splunk::params::forwarder_pkg_ensure,
-  $logging_port           = $splunk::params::logging_port,
-  $splunkd_port           = $splunk::params::splunkd_port,
-  $install_options        = $splunk::params::forwarder_install_options,
-  $splunk_user            = $splunk::params::splunk_user,
-  $splunkd_listen         = '127.0.0.1',
-  $purge_deploymentclient = false,
-  $purge_inputs           = false,
-  $purge_outputs          = false,
-  $purge_props            = false,
-  $purge_transforms       = false,
-  $purge_web              = false,
-  $pkg_provider           = $splunk::params::pkg_provider,
-  $forwarder_confdir      = $splunk::params::forwarder_confdir,
-  $forwarder_output       = $splunk::params::forwarder_output,
-  $forwarder_input        = $splunk::params::forwarder_input,
-  $create_password        = $splunk::params::create_password,
-  $addons                 = {},
+  String $server                        = $splunk::params::server,
+  Boolean $manage_package_source        = true,
+  Optional[String] $package_source      = undef,
+  String $package_name                  = $splunk::params::forwarder_pkg_name,
+  String $package_ensure                = $splunk::params::forwarder_pkg_ensure,
+  Variant[String,Integer] $logging_port = $splunk::params::logging_port,
```

Comment:
I wouldn't be against being more aggressive with these type changes even if it'd mean a breaking change.
eg using `Stdlib::Port` here.

Reply:
@bastelfreak Thoughts?

Reply:
@alexjfisher I vote for doing it.

Reply:
:+1:, please add stricter types.

Reply:
I've added stricter types where I thought it was appropriate and updated that weird params class to have types set there too.